agenthub/codeact_agent/codeact_agent.py

```diff
@@ -41,6 +41,58 @@ def parse_response(response) -> str:
     return action
 
 
+def action_to_str(action: Action) -> str:
+    if isinstance(action, CmdRunAction):
+        return f'{action.thought}\n<execute_bash>\n{action.command}\n</execute_bash>'
+    elif isinstance(action, IPythonRunCellAction):
+        return f'{action.thought}\n<execute_ipython>\n{action.code}\n</execute_ipython>'
+    elif isinstance(action, BrowseInteractiveAction):
+        return f'{action.thought}\n<execute_browse>\n{action.browser_actions}\n</execute_browse>'
+    elif isinstance(action, MessageAction):
+        return action.content
+    return ''
+
+
+def get_action_message(action: Action) -> dict[str, str] | None:
+    if isinstance(action, MessageAction) and action.source == 'user':
```

Comment:
It seems this special case for MessageAction can be handled the same way as `elif` does, since it sets the source and then action_to_str() is indeed `content`

Reply:
Good call!